src/main/resources/templates/db_scripts/data-migration/13.8.0/populate_search_locations_ids.sql

```diff
@@ -0,0 +1,29 @@
+-- Update searchLocationIds arrays co contains all location ids and all location holding permanent location ids
+-- 1. Add location ids
+UPDATE ${myuniversity}_${mymodule}.po_line
+   SET jsonb = jsonb || jsonb_build_object(
+         'searchLocationIds',
+         jsonb->'searchLocationIds' || COALESCE(
+                 (SELECT jsonb_agg(loc->>'locationId')
+                    FROM jsonb_array_elements(jsonb->'locations') loc), '[]'::jsonb)
+       );
+
+-- 2. Add holding permanent location ids
+UPDATE ${myuniversity}_${mymodule}.po_line
+   SET jsonb = jsonb || jsonb_build_object(
+         'searchLocationIds',
+         jsonb->'searchLocationIds' || COALESCE(
+                 (SELECT jsonb_agg(h.permanentLocationId)
+                    FROM jsonb_array_elements(jsonb->'locations') loc,
+                         ${myuniversity}_mod_inventory_storage.holdings_record h
+                    WHERE h.id = uuid(loc->>'holdingId')), '[]'::jsonb)
```

Comment:
we can use WITH as assign to variable. It is mainly for easier to read because of simpler version, no performance efficiency can't be achieved. Optional suggestion

Reply:
I had that version coded as well. The size of complexity removed was a very small and we still had an inner SQL with where clause only. I'd consider using `WITH` again with slightly complex join criteria or where clause.

Reply:
OK, skip suggestion